### PR TITLE
[Flax] added broadcast_to_shape_from_left helper and Scheduler tests

### PR DESCRIPTION
**Pull request by kashif**
instead of the while loop

### CODE REVIEW

**Comment by kashif** [comment]
yes let me add flax scheduler tests! good idea!

**Comment by kashif** [comment]
I will integrate #580 here

**Comment by kashif** [comment]
@anton-l i have added some flax scheduler tests... whenever you get a chance i would not mind a review. Thanks!

File: src/diffusers/schedulers/scheduling_ddpm_flax.py [hunk 1]
```diff
@@ -101,6 +101,10 @@ class FlaxDDPMScheduler(FlaxSchedulerMixin, ConfigMixin):
 
     """
 
+    @property
+    def has_state(self):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddpm_flax.py:105]: Why do we need `has_state`?

Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_ddpm_flax.py:105]: I don't see it being called anywhere?

**Comment by patrickvonplaten** [comment]
Did we test the changes on a TPUv3-8 or TPUv2-8?

File: src/diffusers/schedulers/scheduling_ddpm_flax.py (hunk 1 above)
Comment (kashif) [reply src/diffusers/schedulers/scheduling_ddpm_flax.py:105]: yup just made things mode consistent

**Comment by patrickvonplaten** [comment]
Happy to merge once confirmed everything works on TPUv3-8 and fast tests pass. Think the current failure of the fast tests is unrelated and has been fixed by Suraj here: https://github.com/huggingface/diffusers/pull/928

**Comment by kashif** [comment]
@patrickvonplaten thanks let me update my branch!

**Comment by pcuenca** [comment]
The following test passes in a TPU v3-8: https://github.com/huggingface/diffusers/blob/2fdd094c10aadd71c17d3e3d28e47b6c3cf26bd2/tests/test_pipelines_flax.py#L70

But the others in the same class don't. For example, we are now getting `(8, 1, 128, 128, 3)` here: https://github.com/huggingface/diffusers/blob/2fdd094c10aadd71c17d3e3d28e47b6c3cf26bd2/tests/test_pipelines_flax.py#L62. The same thing happens in `main`, but it works in 7c2262640bbf9fa61c281bc49eb8494cb48da81f.

I'm a bit puzzled, I'll take another look when I've had some rest.

**Comment by patrickvonplaten** [comment]
Ok as this PR gives the same results as `main` merging this for now

**Pull request merged**